client/src/Pages/HomePage/index.js

```diff
@@ -0,0 +1,49 @@
+import React, { useState, useEffect } from 'react';
+import axios from 'axios';
+
+import { message } from 'antd';
+import Search from '../../Components/Search';
+import CardContainer from '../../Components/CardContainer';
+import './style.css';
+
+const HomePage = () => {
+  const [kindergartensData, setKindergartensData] = useState([]);
+  const fetchData = async () => {
+    try {
+      const {
+        data: { data },
+      } = await axios.get(`/api/v1/kindergarten`);
+      const sortRes = data.sort((a, b) => {
+        if (a.rating_average < b.rating_average) {
+          return 1;
+        }
+        if (a.rating_average > b.rating_average) {
+          return -1;
+        }
+        return 0;
+      });
+      setKindergartensData(sortRes);
+    } catch (err) {
+      message.error('خطأ في السيرفر يرجى المحاولة لاحقًا.');
+    }
+  };
+
+  useEffect(() => {
+    fetchData();
+  }, []);
```

Comment:
when you make  REST call in useEffects you have to cleanup it
that link will help you to do that:
https://stackoverflow.com/questions/53861916/canceling-an-axios-rest-call-in-react-hooks-useeffects-cleanup-failing